olm/build-bundles.sh

```diff
@@ -0,0 +1,152 @@
+      esac
+      shift
+  done
+
+  # e.g. "airflow" instead of "airflow-operator"
+  OPERATOR=$(echo "${OPERATOR_NAME}" | cut -d- -f1)
+}
+
+bundle-clean() {
+	rm -rf "bundle"
+	rm -rf "bundle.Dockerfile"
+}
+
+build-bundle() {
+	opm alpha bundle generate --directory manifests --package "${OPERATOR_NAME}-package" --output-dir bundle --channels stable --default stable
+  cp metadata/*.yaml bundle/metadata/
+  docker build -t "docker.stackable.tech/stackable/${OPERATOR_NAME}-bundle:${VERSION}" -f bundle.Dockerfile .
+  docker push "docker.stackable.tech/stackable/${OPERATOR_NAME}-bundle:${VERSION}"
+  opm alpha bundle validate --tag "docker.stackable.tech/stackable/${OPERATOR_NAME}-bundle:${VERSION}" --image-builder docker
+}
+
+setup() {
+  if [ -d "catalog" ]; then
+    rm -rf catalog
+  fi
+
+  mkdir -p catalog
+  rm -f catalog.Dockerfile
+  rm -f catalog-source.yaml
+}
+
+catalog() {
+  opm generate dockerfile catalog
+
+  echo "Initiating package: ${OPERATOR}"
+  opm init "stackable-${OPERATOR}-operator" \
+      --default-channel=stable \
+      --description=./README.md \
+      --output yaml > "catalog/stackable-${OPERATOR}-operator.yaml"
+  echo "Add operator to package: ${OPERATOR}"
+  {
+    echo "---"
+    echo "schema: olm.channel"
+    echo "package: stackable-${OPERATOR}-operator"
+    echo "name: stable"
+    echo "entries:"
+    echo "- name: ${OPERATOR}-operator.v${VERSION}"
+  } >> "catalog/stackable-${OPERATOR}-operator.yaml"
+  echo "Render operator: ${OPERATOR}"
+  opm render "docker.stackable.tech/stackable/${OPERATOR}-operator-bundle:${VERSION}" --output=yaml >> "catalog/stackable-${OPERATOR}-operator.yaml"
+
+  echo "Validating catalog..."
+  opm validate catalog
+
+  echo "Build and push catalog for all ${OPERATOR} operator..."
+  docker build . -f catalog.Dockerfile -t "docker.stackable.tech/stackable/stackable-${OPERATOR}-catalog:${VERSION}"
+  docker push "docker.stackable.tech/stackable/stackable-${OPERATOR}-catalog:${VERSION}"
+}
+
+deploy() {
+  if $DEPLOY; then
+    echo "Deploying catalog..."
+
+    {
+    echo "---"
+    echo "apiVersion: operators.coreos.com/v1alpha1"
+    echo "kind: CatalogSource"
+    echo "metadata:"
+    echo "  name: stackable-${OPERATOR}-catalog"
+    echo "  namespace: stackable-operators"
+    echo "spec:"
+    echo "  sourceType: grpc"
+    echo "  image: docker.stackable.tech/stackable/stackable-${OPERATOR}-catalog:${VERSION}"
+    echo "  displayName: Stackable Catalog"
+    echo "  publisher: Stackable GmbH"
+    echo "  updateStrategy:"
+    echo "    registryPoll:"
+    echo "      interval: 10m"
+    } >> catalog-source.yaml
+
+    kubectl apply -f catalog-source.yaml
+    echo "Catalog deployment successful!"
+  fi
+}
+
+main() {
+  parse_inputs "$@"
+  if [ -z "${VERSION}" ] || [ -z "${BRANCH}" ] || [ -z "${OPERATOR_NAME}" ]; then
+    echo "Usage: $SCRIPT_NAME -r <release> -b <branch> -o <operator>"
+    exit 1
+  fi
+
+  rm -rf "/tmp/openshift-bundles"
```

Comment:
Works for me. Just for future projects you could use something like `mktemp -d -t 'openshift-bundles-XXXXXXXX'`, which gives e.g. `/tmp/openshift-bundles-eUCh4ns8`.

Reply:
Thanks! Will incorporate that.